python/hail/methods/statgen.py

```diff
@@ -1721,25 +1723,111 @@ def pc_relate(dataset, k, maf, block_size=512, min_kinship=-float("inf"), statis
+def pc_relate_with_scores(ds, scores, maf, path=None, block_size=512, min_kinship=-float("inf"), statistics="all"):
+    """The PC-Relate method parameterized by sample PC scores
+
+    See the detailed documentation at :meth:`.pc_relate`.
+
+    Parameters
+    ----------
+    ds : :class:`.MatrixTable`
+        A biallelic-variant keyed :class:`.MatrixTable` containing
+        genotype information.
+    scores : :class:`ArrayNumericExpression`
+        A column-indexed expression defining the principal component scores for
+        each column. Every column must have the same number of scores.
+    maf : :obj:`float`
+        The minimum individual-specific allele frequency for an allele used to
+        measure relatedness.
+    path : :obj:`str` or :obj:`None`
+        A temporary directory to store intermediate matrices. Storing the matrices
+        to a file system is necessary for reliable execution of this method.
+    block_size : :obj:`int`
+        the side length of the blocks of the block-distributed matrices; this
+        should be set such that at least three of these matrices fit in memory
+        (in addition to all other objects necessary for Spark and Hail).
+    min_kinship : :obj:`float`
+        Pairs of samples with kinship lower than ``min_kinship`` are excluded
+        from the results.
+    statistics : :obj:`str`
+        the set of statistics to compute, `phi` will only compute the
+        kinship statistic, `phik2` will compute the kinship and
+        identity-by-descent two statistics, `phik2k0` will compute the
+        kinship statistics and both identity-by-descent two and zero, `all`
+        computes the kinship statistic and all three identity-by-descent
+        statistics.
 
+    Returns
+    -------
+    :class:`.Table`
+        A :class:`.Table` mapping pairs of samples to estimations of their
+        kinship and identity-by-descent zero, one, and two.
+
+        The fields of the resulting :class:`.Table` entries are of types: `i`:
+        ``ds.col_key.dtype``, `j`: ``ds.col_key.dtype``, `kin`:
+        :py:data:`.tfloat64`, `k2`: :py:data:`tfloat64`, `k1`:
+        :py:data:`tfloat64`, `k0`: :py:data:`tfloat64`. The table is keyed by
+        `i` and `j`.
     """
-    require_col_key_str(dataset, 'pc_relate')
-    dataset = require_biallelic(dataset, 'pc_relate')
-    intstatistics = {"phi": 0, "phik2": 1, "phik2k0": 2, "all": 3}[statistics]
-    _, scores, _ = hwe_normalized_pca(dataset, k, False, True)
+    assert(scores._indices.source == ds)
+    # analyze('pc_relate_with_scores', scores, ds._col_indices)
+    ds = ds.annotate_cols(scores=scores)
+    ds = require_biallelic(ds, 'pc_relate_with_scores')
+    ds = require_unique_samples(ds, 'pc_relate_with_scores')
+    ds = ds.select_entries(GT=ds.GT,
+                           naa=hl.float64(ds.GT.n_alt_alleles()))
+    ds = ds.select_rows(
+        *ds.row_key,
+        mean_gt=(agg.sum(ds.naa) /
```

Comment:
Do you really even need the 2nd aggregator? `agg.mean()` should handle this fine, right?

Reply:
yes you're right @jbloom22 can you add this to your changes?